docs/tekton-bundle-contracts.md

```diff
@@ -0,0 +1,97 @@
+<!--
+---
+linkTitle: "Tekton Bundle Contracts"
+weight: 8
+---
+-->
+# Tekton Bundle Contract v0.1
+
+When using a Tekton Bundle in a task or pipeline reference, the OCI artifact backing the
+bundle must adhere to the following contract.
+
+## Contract
+
+Only Tekton CRDs (eg, `Task` or `Pipeline`) may reside in a Tekton Bundle used as a Tekton
+bundle reference.
+
+Each layer of the image must map 1:1 with a single Tekton resource (eg Task).
+
+Each layer must contain all of the following annotations:
+
+- `org.opencontainers.image.title` => `ObjectMeta.Name` of the resource
```

Comment:
I don't know if `org.opencontainers.image.title` is used quite correctly here, I might just define your own dev.tekton thing, but it probably doesn't matter.

Reply:
You're right. This is overloading what should be the image name. I will use a custom annotation here as well.